src/ToolBox/superpmi/mcs/verbstrip.cpp

```diff
@@ -32,6 +32,7 @@ int verbStrip::DoWork(
     MethodContextReader* reader = new MethodContextReader(nameOfInput, indexes, indexCount);
     if (!reader->isValid())
     {
+        delete reader;
```

Comment:
It would seem that there's no need to dynamically allocate the reader in the first place.

Reply:
@mikedn Not sure what you're suggesting here. Can you elaborate please?

Reply:
```C++
MethodContextReader reader(nameOfInput, indexes, indexCount); 
if (!reader.isValid())
{
etc.
```
That said, I'm not familiar with this code. It may be that MethodContextReader does weird thing that would prevent this from working correctly (e.g. storing the `this` pointer somewhere).

Reply:
@mikedn Forgive my ignorance, but I don't understand how you're coming to the conclusion that dynamic memory allocation isn't necessary here.

Reply:
The `reader` pointer isn't stored anywhere or returned from the function. Basically all its uses are:
```C++
MethodContextReader* reader = new MethodContextReader(nameOfInput, indexes, indexCount);
if (!reader->isValid())
MethodContextBuffer mcb = reader->GetNextMethodContext();
LogVerbose("%2.1f%% - Loaded %d at %d per second", reader->PercentComplete(), loadedCount, (int)((double)500 / st1->GetSeconds()));
```
So unless `MethodContextReader` does something rather unusual such as storing `this` in a static variable, there's no need for dynamic allocation. The object does not outlive the function so it can be a local variable.

Reply:
@mikedn Thank you so much for this insight. TIL.

Reply:
@mikedn Resolved by switching to static allocation instead.